crates/blockifier/src/execution/syscalls/syscall_tests.rs

```diff
@@ -279,68 +280,136 @@ fn emit_events(
+
+        (state, block_number, block_hash)
+    }
 
-    // Initialize block number -> block hash entry.
-    let upper_bound_block_number = CURRENT_BLOCK_NUMBER - constants::STORED_BLOCK_HASH_BUFFER;
-    let block_number = stark_felt!(upper_bound_block_number);
-    let block_hash = stark_felt!(66_u64);
-    let key = StorageKey::try_from(block_number).unwrap();
-    let block_hash_contract_address =
-        ContractAddress::try_from(StarkFelt::from(constants::BLOCK_HASH_CONTRACT_ADDRESS)).unwrap();
-    state.set_storage_at(block_hash_contract_address, key, block_hash).unwrap();
+    #[test_case(FeatureContract::SierraTestContract, NATIVE_GAS_PLACEHOLDER; "Native")]
+    #[test_case(FeatureContract::TestContract(CairoVersion::Cairo1), 14250; "VM")]
+    fn positive_flow(test_contract: FeatureContract, expected_gas: u64) {
+        let (mut state, block_number, block_hash) = initialize_state(test_contract);
 
-    // Positive flow.
-    let calldata = calldata![block_number];
-    let entry_point_call = CallEntryPoint {
-        entry_point_selector: selector_from_name("test_get_block_hash"),
-        calldata,
-        ..trivial_external_entry_point_new(test_contract)
-    };
+        let calldata = calldata![block_number];
+        let entry_point_call = CallEntryPoint {
+            entry_point_selector: selector_from_name("test_get_block_hash"),
+            calldata,
+            ..trivial_external_entry_point_new(test_contract)
+        };
 
-    assert_eq!(
-        entry_point_call.clone().execute_directly(&mut state).unwrap().execution,
-        CallExecution {
-            gas_consumed: expected_gas,
-            ..CallExecution::from_retdata(retdata![block_hash])
-        }
-    );
+        pretty_assertions::assert_eq!(
+            entry_point_call.clone().execute_directly(&mut state).unwrap().execution,
+            CallExecution {
+                gas_consumed: expected_gas,
+                ..CallExecution::from_retdata(retdata![block_hash])
+            }
+        );
+    }
 
-    assert_consistent_contract_version(test_contract, &state);
+    #[test_case(FeatureContract::SierraTestContract; "Native")]
+    #[test_case(FeatureContract::TestContract(CairoVersion::Cairo1); "VM")]
+    fn negative_flow_execution_mode_validate(test_contract: FeatureContract) {
+        let (mut state, block_number, _) = initialize_state(test_contract);
 
-    // Negative flow. Execution mode is Validate.
-    let execution_result = entry_point_call.execute_directly_in_validate_mode(&mut state).unwrap();
+        let calldata = calldata![block_number];
+        let entry_point_call = CallEntryPoint {
+            entry_point_selector: selector_from_name("test_get_block_hash"),
+            calldata,
+            ..trivial_external_entry_point_new(test_contract)
+        };
 
-    assert_consistent_contract_version(test_contract, &state);
+        if let FeatureContract::TestContract(_) = test_contract {
```

Comment:
I think is better to use the macro `matches!` here. You want to select this branch based on the type and not actually unwrap it